File: mikado-method/test-data-builders-kata/typescript/test/com/murex/tbw/report/CountryBuilder.ts
import { Country, Currency, Language } from "../../../../../src/com/murex/tbw/domain/country";

export class CountryBuilder {
    private name: string = "country";
    private currency: Currency = Currency.Dollar;
    private language: Language = Language.English;

    static aCountry(): CountryBuilder {
        return new CountryBuilder();
    }
    named(name: string): CountryBuilder {
        this.name = name;
        return this;
    }
    payingIn(currency: Currency): CountryBuilder {
        this.currency = currency;
        return this;
    }

    speaking(language: Language): CountryBuilder {
        this.language = language;
        return this;
    }

    build(): Country {
        return new Country(this.name, this.currency, this.language);
    }

    static USA = this.aCountry().named("USA").payingIn(Currency.Dollar).speaking(Language.English);
}

File: mikado-method/test-data-builders-kata/csharp/BookInvoicing.Tests/CountryBuilder.cs
using BookInvoicing.Domain.Country;

namespace BookInvoicing.Tests
{
    public class CountryBuilder
    {
        private string _name = "name";
        private Currency _currency = Currency.Euro;
        private Language _language = Language.French;

        public static CountryBuilder ACountry()
        {
            return new CountryBuilder();
        }

        public static CountryBuilder Usa()
        {
            return ACountry()
                .Named("USA")
                .WhoPaysIn(Currency.UsDollar)
                .Speaking(Language.French);
        }
        
        public CountryBuilder Named(string name)
        {
            _name = name;
            return this;
        }

        public CountryBuilder WhoPaysIn(Currency currency)
        {
            _currency = currency;
            return this;
        }

        public CountryBuilder Speaking(Language language)
        {
            _language = language;
            return this;
        }

        public Country Build()
        {
            return new Country(_name, _currency, _language);
        }

    }
}